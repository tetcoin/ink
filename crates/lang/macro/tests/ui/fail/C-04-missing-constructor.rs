use pro_lang as pro;

#[pro::contract]
mod noop {
    #[pro(storage)]
    pub struct Noop {}

    impl Noop {
        #[pro(message)]
        pub fn noop(&self) {}
    }
}

fn main() {}
