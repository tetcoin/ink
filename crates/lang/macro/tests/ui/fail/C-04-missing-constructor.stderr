error: missing pro! constructor
  --> $DIR/C-04-missing-constructor.rs:4:1
   |
4  | / mod noop {
5  | |     #[pro(storage)]
6  | |     pub struct Noop {}
7  | |
...  |
11 | |     }
12 | | }
   | |_^
